mne/utils/mixin.py

```diff
@@ -501,8 +575,107 @@ def decimate(self, decim, offset=0, verbose=None):
         else:
             self._decim_slice = decim_slice
         self._set_times(self._raw_times[self._decim_slice])
+        self._update_first_last()
+        return self
+
+    # Overridden method signature does not match call...
+    def time_as_index(self, times, use_rounding=False):  # lgtm
```

Comment:
can you comment on this and the below methods, to highlight any changed lines (or confirm that it's a straight copy-paste from elsewhere)?

Reply:
It is a straight copy-paste from `mne.io.base.py`